src/client/App.test.tsx

```diff
@@ -0,0 +1,125 @@
+import type { EditorProps } from "@monaco-editor/react";
+import { TooltipProvider } from "@radix-ui/react-tooltip";
+import {
+	cleanup,
+	render,
+	screen,
+	waitFor,
+} from "@testing-library/react";
+import type { FC } from "react";
+import { createBrowserRouter, RouterProvider } from "react-router";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+import { App } from "@/client/App";
+import { EditorProvider } from "@/client/contexts/editor";
+import { ThemeProvider } from "@/client/contexts/theme";
+import attachGPUExample from "@/examples/code/attach-gpu.tf?raw";
+import defaultExample from "@/examples/code/default.tf?raw";
+import { initWasm } from "@/utils/wasm";
```

Comment:
does this actually work? you're importing it before the mock is set up (I always avoid mocks because this sort of thing is so confusing to me)

Reply:
Yeah because vitest moves all mocks to the top of the file